.github/workflows/repo-check.yml

```diff
@@ -16,16 +16,26 @@ permissions:
 
 on:
   push:
+  pull_request:
+  workflow_dispatch:
+
+# This workflow is run on pushes to any branch in the MFEM repo (with or without
+# PRs), as well as on updates to PRs from forks. In particular, we do not
+# duplicate work by running on both pushes and updates to local PRs. We do that
+# by checking if the workflow trigger is 'push' ("github.event_name == 'push'")
+# and if we are in a fork ("github.event.pull_request.head.repo.full_name !=
+# github.repository").
 
 jobs:
   file-headers-check:
     runs-on: ubuntu-18.04
-
+    if: (github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository)
```

Comment:
It may be better to split long lines like this using the multi-line yaml syntax, e.g. something like this:
```yaml
     if: |
       (github.event_name == 'push' ||
        github.event.pull_request.head.repo.full_name != github.repository)
```
See https://github.community/t/how-to-write-multi-line-condition-in-if/128477.

Reply:
done in https://github.com/mfem/mfem/pull/3001/commits/b30b091ae9f3a4f33be19ca85efa2d8351ca302a